platformio/ide/tpls/clion/CMakeListsPrivate.txt.tpl

```diff
@@ -37,12 +47,16 @@ SET(CMAKE_C_STANDARD {{ cc_stds[-1] }})
 set(CMAKE_CXX_STANDARD {{ cxx_stds[-1] }})
 % end
 
-% for define in defines:
-add_definitions(-D'{{!re.sub(r"([\"\(\)#])", r"\\\1", define)}}')
-% end
+% for env in envs:
+if (CMAKE_BUILD_TYPE MATCHES {{ env }})
+%   items = load_project_ide_data(get_project_dir(),env)
+%   for define in items["defines"]:
+    add_definitions(-D'{{!re.sub(r"([\"\(\)#])", r"\\\1", define)}}')
+%   end
 
-% for include in includes:
-include_directories("{{ _normalize_path(include) }}")
+%   for include in items["includes"]:
+    include_directories("{{ _normalize_path(include) }}")

```

Comment:
As you use raw includes from build env, need to convert path to unix style, CMake does not like `\\` Windows slahes. So, please use ``{{  _normalize_path(to_unix_path(include)) }}``.